server/channels/api4/bleve.go

```diff
@@ -18,16 +18,11 @@ func purgeBleveIndexes(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec := c.MakeAuditRecord("purgeBleveIndexes", audit.Fail)
 	defer c.LogAuditRec(auditRec)
 
-	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionPurgeBleveIndexes) {
+	if !c.App.SessionHasPermissionToCheckRestrictedAdmin(*c.AppContext.Session(), model.PermissionPurgeBleveIndexes) {
```

Comment:
I'm not a fan of mixing configuration settings and permissions: in an ideal world, I imagine we would change the definition of the restricted system admin role to simply not have `model.PermissionPurgeBleveIndexes`.

For expediency, and to minimize thrash, I'm just bringing the existing pattern under a common umbrella (that in turn honours local mode.)

Reply:
> I imagine we would change the definition of the restricted system admin role to simply not have model.PermissionPurgeBleveIndexes.

Indeed. This is a big maintenance headache as this config setting is sprinkled all over the codebase. Bringing everything under a single function is a good first step.